libwnbd/libwnbd.cpp

```diff
@@ -235,6 +235,44 @@ DWORD WnbdRemoveEx(
     return Status;
 }
 
+DWORD WnbdSetDiskSize(PWNBD_DISK Disk, UINT64 BlockCount) {
+    DWORD ErrorCode = ERROR_SUCCESS;
+    DWORD BytesReturned = 0;
+
+    LogDebug("Resizing device %s.",
+        Disk->Properties.InstanceName);
+    if (!Disk->Handle || Disk->Handle == INVALID_HANDLE_VALUE) {
+        LogDebug("WNBD device not found");
+        return ERROR_NOT_FOUND;
+    }
+
+    ErrorCode = WnbdIoctlSetDiskSize(
+        Disk->Handle,
+        Disk->ConnectionInfo.ConnectionId,
+        BlockCount,
+        NULL);
+
+    if (ErrorCode) {
+        LogError("Could not set disk size.");
+        return ErrorCode;
+    }
+
+    if (!DeviceIoControl(
+        Disk->Handle,
+        IOCTL_DISK_UPDATE_PROPERTIES,
+        NULL,
+        0,
+        NULL,
+        0,
+        &BytesReturned,
+        NULL)) {
```

Comment:
nit: let's increase the parameter indentation here so that it's clearly delimited from the ``if`` code block.